crates/testing/tests/queries.rs

```diff
@@ -52,3 +53,126 @@ fn handling_multi_query() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+mod query_super_smart {
+    use {
+        borsh::{BorshDeserialize, BorshSerialize},
+        grug_storage::Item,
+        grug_types::{
+            to_json_value, Addr, Empty, ImmutableCtx, Json, MutableCtx, Response, Uint128,
+        },
+        serde::{Deserialize, Serialize},
+    };
+
+    const CONFIG: Item<ConfigHost> = Item::new("c");
+
+    #[derive(Serialize, Deserialize, BorshDeserialize, BorshSerialize)]
+    pub struct ConfigHost {
+        pub owner: String,
+        pub balance: Uint128,
+    }
+
+    #[derive(Serialize, Deserialize, grug_macros::Query)]
+    #[query_path_override(::grug_types)]
+    pub enum QueryMsgHost {
+        #[returns(ConfigHost)]
+        Config {},
+    }
+
+    pub fn instantiate_host(ctx: MutableCtx, config: ConfigHost) -> anyhow::Result<Response> {
+        CONFIG.save(ctx.storage, &config)?;
+        Ok(Response::new())
+    }
+
+    pub fn query_host(ctx: ImmutableCtx, msg: QueryMsgHost) -> anyhow::Result<Json> {
+        match msg {
+            QueryMsgHost::Config {} => Ok(to_json_value(&CONFIG.load(ctx.storage)?)?),
+        }
+    }
+
+    // ---- Client ----
+    #[derive(Serialize, Deserialize, grug_macros::Query)]
+    #[query_path_override(::grug_types)]
```

Comment:
If you put this test in `crates/std/tests/queries.rs`, you can then import the macro as `grug::Query`, there's no need to define this override.

I made a test for `#[grug::index_list]` this way.